[PORT] High-Frequency Blade.

Moccha-Bee opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Ports the high frequency blade from hippie station.]
from: https://github.com/HippieStation/HippieStation/pull/11767
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Does no damage against objects and just insta delimbs. A more intersting and quirky item for [traitors.](https://cdn.discordapp.com/attachments/427337870844362755/644943403016257537/unknown.png)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: High frequency blade.

/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
[PORT] High-Frequency Blade.

Pots the high frequency blade from hippie station.

MarkSuckerberg commented:
As much as I love the concept, I don't think an instant delimb (which is what this seems to do) seems a bit OP and needs balancing, ESPECIALLY if it's only 8tc.

Moccha-Bee commented:
I agree with you but considering how it dosn't do damage to windows nor anything else a small increment in price like 4tc should be enough to makae it more balanced or 6tc

Kerbin-Fiber commented:
This sword blocks all melee attacks and even has a 1 in 5 chance of delimbing someone when *they* attack you.

Moccha-Bee commented:
> This sword blocks all melee attacks and even has a 1 in 5 chance of delimbing someone when _they_ attack you.

Okay now this is epic will increase the price

Commit:
Update uplink_items.dm

Kerbin-Fiber commented:
This sword has to be at least more than 10tc as if it is 10 or less, you could pair it with an Esword and have a 50% of a blocked projectile ONTOP of the above

Moccha-Bee commented:
Increased the price to 12TC

Ruediger4 commented:
"A one hit delimb is too op"

- Literally can't deflect shots
- Can't kill you
- It's pretty fucking loud like a esword
- Takes up your belt slot to carry it with the sheath, which means you can't hide it like the esword

IMO 9 TC is more than fine. It's a more direct-approach melee weapon that you *can't* hide and doesn't even kill by itself.

Ruediger4 commented:
For reference:

- Esword costs 9 TC and can kill, delimb, and you can hide it in your pocket/a storage implant. Can deflect shots, you can throw it and ember it into someone's arse just fine. Scary to approach, and also a gamble to shoot at since your shot can be deflected and you can disable yourself.

- HF Blade costs the same, can't kill but 100% delimb chance, can't hide the sheath, let alone the sword. Can't deflect shots. Throw it and nothing happens. Direct melee approach means you are in a 20% gamble on delimbing yourself like a twat.

Ruediger4 commented:
> This sword has to be at least more than 10tc as if it is 10 or less, you could pair it with an Esword and have a 50% of a blocked projectile ONTOP of the above

So basically a desword but you need to change hands and delimb first before hittin' them with the sword 3 times instead of outright critting in 3 hits with the desword.

MarkSuckerberg commented:
1 in 5 chance of delimbing anyone who hits you wtf

Ruediger4 commented:
> 1 in 5 chance of delimbing anyone who hits you wtf

It's literally just a "don't approach me" type of weapon. Use ranged, slip, flashbangs, throw shit, anything but a direct melee approach and you're fine. If you actually approach and try to SHOVE a HF Blade user you are asking to get delimbed. Like shoving someone with a gun and not expecting them to shoot you.

MarkSuckerberg commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+	flags_1 = CONDUCT_1
+	obj_flags = UNIQUE_RENAME
+	force = 0.001
+	throwforce = 20
```
Throw it and it does nothing, hmm

Kerbin-Fiber replied:
Throwing it does almost the same damage as a fireaxe, jesus

Moccha-Bee replied:
trowing it is not a really good idea considering how you just trew the only thing that gave you any edge in combat to the enemy to  pick it up and use it against you.

MarkSuckerberg commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 0
+	if(attack_type == MELEE_ATTACK) // It really is not a good idea to get close to this thing.
+		final_block_chance = 100
```
Is that a 100% melee block?

Moccha-Bee replied:
looks like so will reduce it to 20

qwertyquerty replied:
still needs to be reduced.

Kerbin-Fiber commented:
I'd say to bring it down to 10tc and see how lrp fares with this, most of them don't even know a esword has a 50% block chance

Commit:
Update uplink_items.dm

MarkSuckerberg commented:
The (what looks like a) 100% melee block as well as the 1 in 5 chance of it delimbing when the user is hit sounds absolutely crazy to me

Commit:
Update hfblade.dm

Moccha-Bee reviewed (commented):
> The (what looks like a) 100% melee block as well as the 1 in 5 chance of it delimbing when the user is hit sounds absolutely crazy to me

Reduced to 20% block chance

Ruediger4 commented:
![image](https://user-images.githubusercontent.com/57809136/68966889-e0cd8480-07bd-11ea-98b4-a1433a1ab352.png)

Woah!
What's that?
*DIRECT MELEE APPROACH BAD IDEA?!!*
WOAH!!!!

SHIT GUYS LET'S NERF DESWORD, 100% DEFLECT CHANCE AGAISNT LASERS IS KINDA CRAZY!

I'll repeat myself: It is a 'DO NOT APPROACH ME' weapon.

Kerbin-Fiber commented:
- The moment the user slips and someone takes the sword, their dead unless they get it back.
- Any range weapon will get them down, easily.
- Literally just throw shit at the user, the user won't bother picking stuff up as they're busy using their sword.
- Great for killing someone or two, but going loud is gonna make you dead, and the meta will make you valid the moment you have it.

Kerbin-Fiber commented:
I say to keep the 100% melee block chance, that's this weapon's gimmick

Ruediger4 commented:
Leave it at 9 TC, or 10 tops and the 100% melee block chance on. It's literally a different version of the Esword but literally encourages ranged approach rather than outright fucking "ME SHOVE AGAINST GLASS TABLE! ME RUN AT GUY WITH BASEBALL BAT AND OUTROBUST!"

Kerbin-Fiber commented:
This is the opposite of the Desword, which has 100% energy projectile block chance (HoS's tazer can't hit em. but has a really low melee block chance, if not, none at all

Moccha-Bee commented:
Tbh having the sword be 100% block chance just turns the user  target to multiple ranged attacks and slips and if they are dumb enough to trow the weapon at someone  they deserve to get delimbed by their own weapon when someone inevitably picks it up.

Commit:
Update hfblade.dm

Kerbin-Fiber commented:
I agree with rue

Ruediger4 commented:
> Tbh having the sword be 100% block chance just turns the user target to multiple ranged attacks and slips and if they are dumb enough to trow the weapon at someone they deserve to get delimbed by their own weapon when someone inevitably picks it up.

This. So much this.

If you're gonna use it, either you know what you're doing or you just want to have fun. Trying to counter it with direct melee means you deserve to be made a potato and buckled to a chair and shamed by everyone, including the clown.

- Going loud is what is going to inevitably happen, so you'll want to try and approach the ranged guys somehow
- Still vulnerable to throwable weapons like spears and bolas
- Still can get slipped
- Throwing your weapon literally plays Curb Your Enthusiasm as you get delimbed and thrown into space
- You can ignore melee guys and just focus on the ranged ones, but melee guys can still try and find a way to just dunk on you by slipping, throwin shit at you, or just putting themselves between you and the ranged lad.
- Still vulnerable to flashes and flashbangs unless you got protection
- Still vulnerable to chem nades, syringes, smoke machines

Ruediger4 commented:
also @Moccha-Bee papa bless you for porting this shit

Moccha-Bee commented:
> also @Moccha-Bee papa bless you for porting this shit

more to come nyesss

Ruediger4 commented:
> > also @Moccha-Bee papa bless you for porting this shit
> 
> more to come nyesss

Fuck yes. Also nanosuit kinda op but lookin forward to it

Moccha-Bee commented:
> > > also @Moccha-Bee papa bless you for porting this shit
> > 
> > 
> > more to come nyesss
> 
> Fuck yes. Also nanosuit kinda op but lookin forward to it

I already ported it to my own fork but i did a stinky and its on the same branch as this pr.

Ruediger4 commented:
> > > > also @Moccha-Bee papa bless you for porting this shit
> > > 
> > > 
> > > more to come nyesss
> > 
> > 
> > Fuck yes. Also nanosuit kinda op but lookin forward to it
> 
> I already ported it to my own fork but i did a stinky and its on the same branch as this pr.

Based. Modified anything?

MarkSuckerberg reviewed (approved):
Approving the code, not the concept.

Ruediger4 commented:
> Approving the code, not the concept.

Explain for lvl 1 virgin like me

Does this mean you're adding the HF blade yet not approving it or

Moccha-Bee commented:
> > > > > also @Moccha-Bee papa bless you for porting this shit
> > > > 
> > > > 
> > > > more to come nyesss
> > > 
> > > 
> > > Fuck yes. Also nanosuit kinda op but lookin forward to it
> > 
> > 
> > I already ported it to my own fork but i did a stinky and its on the same branch as this pr.
> 
> Based. Modified anything?

Yes i added the emag pr to it and there a few other prs to it but i will take a look on them later on.

MarkSuckerberg commented:
Not a maintainer bb, just saying code is gud for ike and qwerty

Ruediger4 commented:
> > > > > > also @Moccha-Bee papa bless you for porting this shit
> > > > > 
> > > > > 
> > > > > more to come nyesss
> > > > 
> > > > 
> > > > Fuck yes. Also nanosuit kinda op but lookin forward to it
> > > 
> > > 
> > > I already ported it to my own fork but i did a stinky and its on the same branch as this pr.
> > 
> > 
> > Based. Modified anything?
> 
> Yes i added the emag pr to it and there a few other prs to it but i will take a look on them later on.

Oh fuck yes I read about the emag PR, that's fucking good

SomeAngryMiner commented:
i would certainly like more traitor items on the uplink, but having a one shot nugget-blade doesnt sound *quite right* to me, besides the "come to brazil" meme thing by using a multitool.

Moccha-Bee commented:
> i would certainly like more traitor items on the uplink, but having a one shot nugget-blade doesnt sound _quite right_ to me, besides the "come to brazil" meme thing by using a multitool.

It only removes one limb per attack and considering how if you are attacking from range you could just pop a few shotgun slugs or flash and the fight is over you nugget the attacker instead.

Toxici11i commented:
This item is extremely buggy on hippie, do not port it unless you want sanity issues. It does NOT block meele, and it does NOT block disarming unlike what it was intended to

qwertyquerty commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	flags_1 = CONDUCT_1
+	obj_flags = UNIQUE_RENAME
+	force = 0.001
```
???

MarkSuckerberg replied:
Only delimbs, does no damage whatsoever

Moccha-Bee replied:
It's the weapons gimmick.

qwertyquerty replied:
Then why not just make it 0

qwertyquerty commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+	AddComponent(/datum/component/butchering, 30, 95, 5)
+	set_light(brightness)
+	START_PROCESSING(SSobj, src)
+	. = ..()
```
Wh- why are we running ..() twice

Moccha-Bee replied:
changed.

qwertyquerty commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+		playsound(C,pick('sound/misc/desceration-01.ogg','sound/misc/desceration-02.ogg','sound/misc/desceration-01.ogg') ,50, 1, -1)
+		C.adjustBruteLoss(-15) //dismembering a limb deals 15 brute, this heals you instantly after that
+		user.changeNext_move(CLICK_CD_CLICK_ABILITY)
+		return ..()
```
This is obsolete and can be removed.

qwertyquerty commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+		var/hideandseeklogic = rand(1,5) // YES, ADD MORE RNG TO COMBAT.
+		if(hideandseeklogic == 1 || HAS_TRAIT(attacker, TRAIT_CLUMSY))
+			attacker.visible_message("<span class = 'warning'>[attacker] attempts to attack [owner] and accidentally slices their arm off! What an idiot!</span>")
+			var/which_hand = BODY_ZONE_L_ARM
```
These 3 lines can be simplified to `var/which_hand = (attacker.active_hand_index % 2) ? BODY_ZONE_L_ARM : BODY_ZONE_R_ARM`

Moccha-Bee replied:
Changed, thanks for the input.

Kerbin-Fiber commented:
@Moccha-Bee 
Test if it works as intended to see what Toxic is talking about

Moccha-Bee commented:
> @Moccha-Bee
> Test if it works as intended to see what Toxic is talking about

will do.

qwertyquerty commented on code/game/objects/items/hfblade.dm:
```diff
@@ -0,0 +1,142 @@
+	. = ..()
+	AddComponent(/datum/component/butchering, 30, 95, 5)
+	set_light(brightness)
+	START_PROCESSING(SSobj, src)
```
Not sure why we are processing when we never define process(), do you know @ike709 ?

ike709 replied:
No idea. Why would a glorified sword need to process at all anyways?

qwertyquerty replied:
Please remove all processing as it doesn't appear to be needed

Commit:
Update hfblade.dm

Moccha-Bee reviewed (commented):
> @Moccha-Bee
> Test if it works as intended to see what Toxic is talking about

Tested it and it's working but not all the time, its not a 100% melee block chance nor disarm.

Toxici11i commented:
it was originally meant to be undisarmable

Moccha-Bee commented:
> it was originally meant to be undisarmable

Tbh its better this way it has a small chance to remove one of your limbs in case you try to disarm.

Toxici11i commented:
> > it was originally meant to be undisarmable
> 
> Tbh its better this way it has a small chance to remove one of your limbs in case you try to disarm.

no its not, being able to be pushed over and hardstunned for 3-4 seconds while the guy can just grab your sword and chop off your limbs in half that time makes the item useless

Moccha-Bee commented:
> > > it was originally meant to be undisarmable
> > 
> > 
> > Tbh its better this way it has a small chance to remove one of your limbs in case you try to disarm.
> 
> no its not, being able to be pushed over and hardstunned for 3-4 seconds while the guy can just grab your sword and chop off your limbs in half that time makes the item useless

Its your fault for not being prepared to hand the situations you could be thrown in with such a weapon, Its the same thing as getting flashed and getting delimbed with it, your fault no one elses.

Toxici11i commented:
> > > > it was originally meant to be undisarmable
> > > 
> > > 
> > > Tbh its better this way it has a small chance to remove one of your limbs in case you try to disarm.
> > 
> > 
> > no its not, being able to be pushed over and hardstunned for 3-4 seconds while the guy can just grab your sword and chop off your limbs in half that time makes the item useless
> 
> Its your fault for not being prepared to hand the situations you could be thrown in with such a weapon, Its the same thing as getting flashed and getting delimbed with it, your fault no one elses.

having you instantly drop a weapon just because somebody clicks on you once is baton level, the entire reason it was implemented this way was because it was designed to be balanced against RNG stuns.
Having a RNG mechanic in the weapon that is unreliable, and could lead to ided because it failed to perform as expected is fucking gay

Moccha-Bee commented:
> > > > > it was originally meant to be undisarmable
> > > > 
> > > > 
> > > > Tbh its better this way it has a small chance to remove one of your limbs in case you try to disarm.
> > > 
> > > 
> > > no its not, being able to be pushed over and hardstunned for 3-4 seconds while the guy can just grab your sword and chop off your limbs in half that time makes the item useless
> > 
> > 
> > Its your fault for not being prepared to hand the situations you could be thrown in with such a weapon, Its the same thing as getting flashed and getting delimbed with it, your fault no one elses.
> 
> having you instantly drop a weapon just because somebody clicks on you once is baton level, the entire reason it was implemented this way was because it was designed to be balanced against RNG stuns.
> Having a RNG mechanic in the weapon that is unreliable, and could lead to ided because it failed to perform as expected is fucking gay

you bought a weapon knowing full well the risks that came with it, if you play stupid games you win stupid prizes.

Commit:
Update hfblade.dm

Commit:
Update hfblade.dm

ike709 commented:
Stale.
